### PR TITLE
Add ARMv8 Linux cross compile support

### PR DESCRIPTION
**Pull request by powderluv**
Backports a fix in libpng from upstream. 
Adds an aarch64 toolchain file and a cmake script to build.

TEST=run ./cmake_aarch64_gnu.sh and make from build_linux_aarch64
Verified the binaries and libs are ARM 64 bit.

### CODE REVIEW

File: platforms/linux/aarch64-gnu.toolchain.cmake [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-undefined -Wl,--gc-sections -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now ${CMAKE_SHARED_LINKER_FLAGS}")
+set(CMAKE_MODULE_LINKER_FLAGS "-Wl,--no-undefined -Wl,--gc-sections -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now ${CMAKE_MODULE_LINKER_FLAGS}")
+set(CMAKE_EXE_LINKER_FLAGS    "-Wl,--no-undefined -Wl,--gc-sections -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now ${CMAKE_EXE_LINKER_FLAGS}")
+
+#if(USE_NEON)
+#  message(WARNING "You use obsolete variable USE_NEON to enable NEON instruction set. Use -DENABLE_NEON=ON instead." )
+#  set(ENABLE_NEON TRUE)
+#elseif(USE_VFPV3)
+#  message(WARNING "You use obsolete variable USE_VFPV3 to enable VFPV3 instruction set. Use -DENABLE_VFPV3=ON instead." )
+#  set(ENABLE_VFPV3 TRUE)
```
Comment (alalek) [inline platforms/linux/aarch64-gnu.toolchain.cmake]: What is kind of problems with NEON on aarch64?

Comment (powderluv) [inline platforms/linux/aarch64-gnu.toolchain.cmake]: I can remove the commented code block. Essentially when compiling png with its neon optimizations (when ENABLE_NEON is set) it was breaking on AARCH64. Once this patch is in I can dig further into cleaning up AARCH64 + ENABLE_NEON.

**Comment by vpisarev** [comment]
I think, it's much more preferable to turn off neon in libpng only

**Comment by powderluv** [comment]
Since VFP and Neon are standard in ARMv8 I will remove those comments and submit an update PR

**Comment by powderluv** [comment]
Does the new patch look good ? Thx

**Comment by alalek** [comment]
@powderluv Could you please take a look on #6737? Does it work for you? (I tested it on Ubuntu 14.04 + gcc-4.8-aarch64-linux-gnu package)

**Comment by vpisarev** [comment]
this PR will be recreated because of CI issues

**Comment by powderluv** [comment]
you will still need the upstream libpng backport in 3rdparty/libpng/pngpriv.h

**Pull request closed without merging**